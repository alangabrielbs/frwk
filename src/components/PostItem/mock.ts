import { PostItemProps } from '.'

export default {
  userId: 1,
  id: 1,
  title:
    'sunt aut facere repellat provident occaecati excepturi optio reprehenderit',
  body:
    'lorem in reprehenderit in voluptate velit esse quam nihil molestiae consequatur'
} as PostItemProps
