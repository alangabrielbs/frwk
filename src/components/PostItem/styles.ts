import styled, { css } from 'styled-components'

export const Wrapper = styled.article`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.medium};
    border-bottom: 1px solid ${theme.colors.divider};

    a {
      text-decoration: none;
    }

    & + & {
      margin-top: ${theme.spacings.medium};
    }

    :hover {
      ${Title} {
        background-image: ${theme.colors.gradients.primary};
        color: transparent;
        -webkit-background-clip: text;
        background-clip: text;
      }
    }
  `}
`

export const Title = styled.h1`
  ${({ theme }) => css`
    font-weight: ${theme.font.weight.bold};
    font-size: ${theme.font.heading.small};
    line-height: 3.3rem;

    color: ${theme.colors.white};
  `}
`

export const Description = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.content.large};
    line-height: 2rem;
    margin-top: ${theme.spacings.xxsmall};

    color: ${theme.colors.secondaryText};
  `}
`
