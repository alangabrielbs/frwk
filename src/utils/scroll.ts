export const scrollTop = (top = 0) => {
  window.scrollTo({ top: top, behavior: 'smooth' })
}
