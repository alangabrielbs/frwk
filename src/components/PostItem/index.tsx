import Link from 'next/link'

import * as S from './styles'

export type PostItemProps = {
  userId: number
  id: number
  title: string
  body: string
}

const PostItem = ({ body, id, title }: PostItemProps) => (
  <S.Wrapper>
    <Link href={String(id)}>
      <a>
        <S.Title>{title}</S.Title>
        <S.Description>{body}</S.Description>
      </a>
    </Link>
  </S.Wrapper>
)

export default PostItem
