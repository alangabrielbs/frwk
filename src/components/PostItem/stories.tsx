import { Story, Meta } from '@storybook/react'
import { Container } from 'components/Container'

import PostItem, { PostItemProps } from '.'
import mock from './mock'

export default {
  title: 'Components/PostItem',
  component: PostItem,
  args: {
    ...mock
  }
} as Meta

export const Desktop: Story<PostItemProps> = (args) => (
  <Container style={{ marginTop: 40 }}>
    <PostItem {...args} />
  </Container>
)

export const Mobile: Story<PostItemProps> = (args) => (
  <Container style={{ marginTop: 40 }}>
    <PostItem {...args} />
  </Container>
)

Mobile.parameters = {
  layout: 'fullscreen',
  viewport: {
    defaultViewport: 'mobile'
  }
}
