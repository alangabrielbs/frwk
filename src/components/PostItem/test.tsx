import { render, screen } from 'utils/test-utils'

import PostItem from '.'
import mock from './mock'

describe('<PostItem />', () => {
  it('should render correctly', () => {
    const { container } = render(<PostItem {...mock} />)

    expect(
      screen.getByRole('heading', { name: mock.title })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
