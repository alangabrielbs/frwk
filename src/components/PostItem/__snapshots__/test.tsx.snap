// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostItem /> should render correctly 1`] = `
.c1 {
  padding-bottom: 3.2rem;
  border-bottom: 1px solid #323238;
}

.c1 a {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 + .c0 {
  margin-top: 3.2rem;
}

.c1:hover .c2 {
  background-image: linear-gradient(110.3deg,#F85D92 8.1%,#6B81FA 91.09%);
  color: transparent;
  -webkit-background-clip: text;
  background-clip: text;
}

.c3 {
  font-weight: 700;
  font-size: 2.8rem;
  line-height: 3.3rem;
  color: #ffffff;
}

.c4 {
  font-size: 1.8rem;
  line-height: 2rem;
  margin-top: 0.8rem;
  color: #A8A8B3;
}

<article
  class="c0 c1"
>
  <a
    href="/1"
  >
    <h1
      class="c2 c3"
    >
      sunt aut facere repellat provident occaecati excepturi optio reprehenderit
    </h1>
    <p
      class="c4"
    >
      lorem in reprehenderit in voluptate velit esse quam nihil molestiae consequatur
    </p>
  </a>
</article>
`
